### ISSUE TITLE
Unable to run `main.py`: "No module named 'tensorflow.python.types'"

### ISSUE THREAD

**Issue by EricCousineau-TRI** [DanielTakeshi/deformable-ravens#4]
I followed the instructions (I think):
```sh
$ cd deformable-ravens
$ git log -n 1 --oneline --no-decorate
6ff2443 remove outdated bag files (we're using the other ones)
$ conda create -n py3-bullet python=3.7
$ conda activate py3-bullet
$ ./install_python_ubuntu.sh
```

Then I got the following error:
```sh
$ python main.py -h
pybullet build time: Sep 22 2020 00:55:20
Detected TensorFlow version:  2.2.0
Traceback (most recent call last):
  File "./main.py", line 45, in <module>
    from ravens import Dataset, Environment, agents, tasks
  File ".../deformable-ravens/ravens/__init__.py", line 2, in <module>
    import ravens.agents as agents
  File ".../deformable-ravens/ravens/agents/__init__.py", line 1, in <module>
    from ravens.agents.dummy import DummyAgent
  File ".../deformable-ravens/ravens/agents/dummy.py", line 10, in <module>
    from ravens.models import Attention, Transport
  File ".../deformable-ravens/ravens/models/__init__.py", line 6, in <module>
    from ravens.models.conv_mlp import ConvMLP
  File ".../deformable-ravens/ravens/models/conv_mlp.py", line 5, in <module>
    import tensorflow_hub as hub
  File "{home}/.local/opt/miniconda3/envs/py3-bullet/lib/python3.7/site-packages/tensorflow_hub/__init__.py", line 29, in <module>
    from tensorflow_hub.estimator import LatestModuleExporter
  File "{home}/.local/opt/miniconda3/envs/py3-bullet/lib/python3.7/site-packages/tensorflow_hub/estimator.py", line 64, in <module>
    class LatestModuleExporter(tf_v1.estimator.Exporter):
  File "{home}/.local/opt/miniconda3/envs/py3-bullet/lib/python3.7/site-packages/tensorflow/python/util/lazy_loader.py", line 62, in __getattr__
    module = self._load()
  File "{home}/.local/opt/miniconda3/envs/py3-bullet/lib/python3.7/site-packages/tensorflow/python/util/lazy_loader.py", line 45, in _load
    module = importlib.import_module(self.__name__)
  File "{home}/.local/opt/miniconda3/envs/py3-bullet/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "{home}/.local/opt/miniconda3/envs/py3-bullet/lib/python3.7/site-packages/tensorflow_estimator/__init__.py", line 10, in <module>
    from tensorflow_estimator._api.v1 import estimator
  File "{home}/.local/opt/miniconda3/envs/py3-bullet/lib/python3.7/site-packages/tensorflow_estimator/_api/v1/estimator/__init__.py", line 10, in <module>
    from tensorflow_estimator._api.v1.estimator import experimental
  File "{home}/.local/opt/miniconda3/envs/py3-bullet/lib/python3.7/site-packages/tensorflow_estimator/_api/v1/estimator/experimental/__init__.py", line 10, in <module>
    from tensorflow_estimator.python.estimator.canned.dnn import dnn_logit_fn_builder
  File "{home}/.local/opt/miniconda3/envs/py3-bullet/lib/python3.7/site-packages/tensorflow_estimator/python/estimator/canned/dnn.py", line 31, in <module>
    from tensorflow_estimator.python.estimator import estimator
  File "{home}/.local/opt/miniconda3/envs/py3-bullet/lib/python3.7/site-packages/tensorflow_estimator/python/estimator/estimator.py", line 51, in <module>
    from tensorflow_estimator.python.estimator import model_fn as model_fn_lib
  File "{home}/.local/opt/miniconda3/envs/py3-bullet/lib/python3.7/site-packages/tensorflow_estimator/python/estimator/model_fn.py", line 29, in <module>
    from tensorflow.python.types import core
ModuleNotFoundError: No module named 'tensorflow.python.types'
```

See output of `conda list --export` here:
https://gist.github.com/EricCousineau-TRI/50284aba406e965956a2827de37745b1

**Comment by EricCousineau-TRI** [comment]
And forgot to say - super excited to give this a whirl!!!

**Comment by DanielTakeshi** [comment]
Thanks @EricCousineau-TRI 
Just to be clear, what was the command that you ran which resulted in the error above? Was it `python main.py [something]`?

**Comment by EricCousineau-TRI** [comment]
Howdy Daniel! It was just `python main.py -h`, and it failed on import.

**Comment by PeterQiu0516** [comment]
Exactly same issue here, on Linux 20.04.

**Comment by PeterQiu0516** [comment]
I am able to resolve the issue by reintalling the tensorflow:
```bash
conda uninstall tensorflow
conda install tensorflow
```

Hope this fix your issue as well @EricCousineau-TRI :)

**Comment by EricCousineau-TRI** [comment]
Yup, that works! Seems like just an upgrade - see ref'd PR #6

**Comment by DanielTakeshi** [comment]
Thanks for reporting @EricCousineau-TRI   I can reproduce your error if I follow the same installation instructions as on the README. I am not sure what happened here ... I definitely ran a clean installation back when I released this code base. I also was able to follow these instructions earlier in 2021 (January/February) when installing this package, so there must have been something that happened since then. The confusing thing is that the installation I use restricts it to certain package versions, so presumably that wouldn't be the issue.

My system is Ubuntu 18.04, by the way.

**Comment by DanielTakeshi** [comment]
Thanks @PeterQiu0516 (and @EricCousineau-TRI  again) for suggesting the upgrade to TensorFlow 2.4.1.

The one thing that I'm cautious about is whether this affects any code that uses TensorFlow addons.  I remember at one point when working on this research project, I had to be careful about using specific versions of TensorFlow that would allow proper usage of the addons package. This is only used for image rotations, and a quick search means it's used in these three areas of the code (and nowhere else):

https://github.com/DanielTakeshi/deformable-ravens/blob/6ff2443ba7f6673ba4696484e052441262cc14d7/ravens/models/attention.py#L60-L76

and 

https://github.com/DanielTakeshi/deformable-ravens/blob/6ff2443ba7f6673ba4696484e052441262cc14d7/ravens/models/transport.py#L99-L119

and

https://github.com/DanielTakeshi/deformable-ravens/blob/6ff2443ba7f6673ba4696484e052441262cc14d7/ravens/models/transport_goal.py#L94-L102

However, since Tensorflow 2.2 clearly isn't working as you both identified, we might as well try with the later version of TensorFlow, and as long as tensorflow-addons behaves the same way, it should be OK.

**Comment by DanielTakeshi** [comment]
Ah, actually maybe the reason why I had TensorFlow 2.2 listed in the installation script is that on Google cloud machines (which I used when I worked on this project) I was using pip install tensorflow and did that for 2.2. But, I think on my local machines where I was getting [the usual CUDA incompatibility nightmares](https://towardsdatascience.com/tensorflow-gpu-installation-made-easy-use-conda-instead-of-pip-52e5249374bc), I went for a simpler conda install. But, for that, I actually needed TensorFlow 2.3.0.

I'm going to merge the pull request from #6 , I think that's the easiest way going forward.
